common/csv_utils.go

```diff
@@ -18,3 +18,25 @@ func ReadCSV(path string) ([][]string, error) {
 
 	return csv.NewReader(csvFile).ReadAll()
 }
+
+func SaveCSV(path string, records [][]string) {
+	if _, err := os.Stat(path); os.IsExist(err) {
+		os.Remove(path)
+	}
+
+	file, err := os.Create(path)
```

Comment:
Correctly handle errors, logging is good, but you should also stop the process and return the error